contracts/mock/MockKlerosLiquid.sol

```diff
@@ -0,0 +1,385 @@
+        uint256 numberOfChoices;
+        Period period; // The current period of the dispute.
+        uint256 lastPeriodChange; // The last time the period was changed.
+        bool ruled; // True if the ruling has been executed, false otherwise.
+        uint256 __appealCount;
+        mapping(uint256 => uint256) __appealToJurorCount;
+    }
+
+    uint256 public constant MIN_JURORS = 3; // The global default minimum number of jurors in a dispute.
+
+    uint256 public constant NON_PAYABLE_AMOUNT = (2**256 - 2) / 2; // An amount higher than the supply of ETH.
+
+    Court[] private __courts;
+
+    uint256 private __disputeCount;
+
+    mapping(uint256 => Dispute) private __disputeIdToDispute;
+
+    mapping(uint256 => uint256) public currentRuling; // Mocks currentRuling() from KlerosLiquid
+
+    modifier requireArbitrationFee(bytes calldata _extraData) {
+        require(
+            msg.value >= arbitrationCost(_extraData),
+            "Not enough ETH to cover arbitration costs."
+        );
+        _;
+    }
+
+    modifier requireAppealFee(uint256 _disputeID, bytes calldata _extraData) {
+        require(
+            msg.value >= appealCost(_disputeID, _extraData),
+            "Not enough ETH to cover appeal costs."
+        );
+        _;
+    }
+
+    modifier onlyDuringPeriod(uint256 _disputeID, Period _period) {
+        require(__disputeIdToDispute[_disputeID].period == _period);
+        _;
+    }
+
+    constructor() {
+        // https://etherscan.io/address/0x988b3A538b618C7A603e1c11Ab82Cd16dbE28069#readContract
+        // General court
+        __courts.push(
+            Court({
+                parent: 0,
+                hiddenVotes: false,
+                feeForJuror: 25000000000000000,
+                jurorsForCourtJump: 511,
+                timesPerPeriod: [
+                    uint256(280800),
+                    uint256(583200),
+                    uint256(583200),
+                    uint256(388800)
+                ]
+            })
+        );
+        // Blockchain court
+        __courts.push(
+            Court({
+                parent: 0,
+                hiddenVotes: false,
+                feeForJuror: 25000000000000000,
+                jurorsForCourtJump: 63,
+                timesPerPeriod: [
+                    uint256(280800),
+                    uint256(583200),
+                    uint256(583200),
+                    uint256(388800)
+                ]
+            })
+        );
+    }
+
+    function changeSubcourtTimesPerPeriod(
+        uint96 _subcourtID,
+        uint256[4] calldata _timesPerPeriod
+    ) external {
+        __courts[_subcourtID].timesPerPeriod = _timesPerPeriod;
+    }
+
+    function __setCurrentRuling(uint256 disputeId, uint256 ruling) external {
+        currentRuling[disputeId] = ruling;
+    }
+
+    function passPeriod(uint256 _disputeID) external {
```

Comment:
Ah ok so if I understand correctly, my [`give-dispute-ruling`](https://github.com/api3dao/api3-dao-dashboard/blob/main/hardhat/hardhat.config.ts#L209) Hardhat task would update to:

1. set zero period lengths for the first couple of periods (the ones preceding the `appeal` period)
2. call `__setCurrentRuling()` to give the ruling
3. call `passPeriod()` a couple of times to get into the `appeal` period

Does that sound about right?

Reply:
Yeah. I figured calling `passPeriod()` once in a while won't be too much of a hassle.